lang/python/numpy/Makefile

```
@@ -0,0 +1,48 @@
+#
+# Copyright (C) 2019 Jakub Piotr Cłapa <jpc@loee.pl>
+# Copyright (C) 2020 Alexandru Ardelean <ardeleanalex@gmail.com>
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=numpy
+PKG_VERSION:=1.18.5
+PKG_RELEASE:=1
+
+PYPI_NAME:=$(PKG_NAME)
+PKG_HASH:=34e96e9dae65c4839bd80012023aadd6ee2ccb73ce7fdf3074c62f301e63120b
+
+PKG_MAINTAINER:=Alexandru Ardelean <ardeleanalex@gmail.com>
+
+PKG_LICENSE:=BSD-3-Clause
+PKG_LICENSE_FILES:=LICENSE.txt
+PKG_CPE_ID:=cpe:/a:numpy:numpy
+
+# yes, zip... sigh
+PYPI_SOURCE_EXT:=zip
+HOST_PYTHON3_PACKAGE_BUILD_DEPENDS="Cython==0.29.19"
```

```suggestion
HOST_PYTHON3_PACKAGE_BUILD_DEPENDS="Cython>=0.29.14"
```
From [INSTALL.rst.txt](https://github.com/numpy/numpy/blob/v1.18.5/INSTALL.rst.txt#prerequisites).

---

umm, i would still lock it at a certain version;
no preference which;
while i was reading for LXML: https://lxml.de/build.html#cython   it kinda made me paranoid about allowing \_\_any\_\_ version to be  `>= __version_x_y__`
they're saying:
```
We even encourage you to not install Cython for a normal release build,
as the generated C code can vary quite heavily between Cython versions,
which may or may not generate correct code for lxml. The pre-generated 
release sources were tested and therefore are known to work.
```

so, for numpy they're not that strict, but i versions being updated underneath me, which i can't control, sounds a bit familiar;
also, if the scope of reproduce-able builds comes back around, a fixed cython version makes this build a bit more reproduce-able;

thoughts/opinions?

---

I think reproducibility is a good thing, but if this dependency is locked at a specific version, I doubt it will be updated with any regularity (if at all) when there are bug fixes, upgrades, etc. (Especially since there will be no uscan emails notifying of upgrades.) For instance, 0.29.20 is already available.

I would prefer updated but not reproducible builds over reproducible but outdated (and potentially buggy) builds, but this is just my personal preference.

I suppose if you add a comment near `PKG_VERSION` reminding us that cython should be checked for upgrades as well, that could also be sufficient.

---

> I would prefer updated but not reproducible builds over reproducible but outdated (and potentially buggy) builds, but this is just my personal preference.

we can argue the buggy element both ways; i've had issues where `new` was `buggy`;
so, there was X that was working, it updated underneath me, and X + 1 was buggy;
had to wait for X + 2, to be fixed again, while temporarily patching X+1 to work;

> I suppose if you add a comment near `PKG_VERSION` reminding us that cython should be checked for upgrades as well, that could also be sufficient.


though, one thing that could work, is to have a global `CYTHON_HOST:="Cython>=0.29.14"` variable, as a sort of global version switch for all Cython host builds;
and keep it in `python-package.mk`, or  `cython.mk` or whichever;

the obvious benefit of this is uniformity for Cythonized builds; and better global control over updates, for updating locked versions;



> I think reproducibility is a good thing, but if this dependency is locked at a specific version, I doubt it will be updated with any regularity (if at all) when there are bug fixes, upgrades, etc. (Especially since there will be no uscan emails notifying of upgrades.) For instance, 0.29.20 is already available.

coming back to this;
i really liked the reproduce-ability project;
too bad, no more effort was spent on this;
and i tried to help with it; i even sent some patches to upstream Python for this;
so, i would like to keep things reproce-able as much as possible;

---

I think "working" and "not working" are not binary states. A version of cython that works with numpy may not work with lxml; a version that works with both may not work for another package. Within one package, a version that works with feature A may not work with feature B. (I don't think simple run-tests actually tell us if a package is fully functional.)

I don't want to over-complicate this (for now anyway :joy:). I'm fine with your choice of `HOST_PYTHON3_PACKAGE_BUILD_DEPENDS` :smile: